sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruneFileSourcePartitionsSuite.scala

```diff
@@ -108,4 +109,54 @@ class PruneFileSourcePartitionsSuite extends QueryTest with SQLTestUtils with Te
       }
     }
   }
+
+  test("SPARK-28169: Convert scan predicate condition to CNF") {
```

Comment:
I'm thinking about adding a base test `PartitionPruningSuiteBase` with some common test cases. Then we can have a `FileSourcePartitionPruningSuite` with file-source specific tests, and `HiveTablePartitionPruningSuite` with hive-table specific tests.

Reply:
> I'm thinking about adding a base test `PartitionPruningSuiteBase` with some common test cases. Then we can have a `FileSourcePartitionPruningSuite` with file-source specific tests, and `HiveTablePartitionPruningSuite` with hive-table specific tests.

Current test in  `FileSourcePartitionPruningSuite ` and  `HiveTablePartitionPruningSuite ` seems don't have common test,  can you show me some point to do these and I will work on this

Reply:
@cloud-fan  Any advise?

Reply:
at least this new test can be shared between the 2 test suites?

Reply:
> at least this new test can be shared between the 2 test suites?

Updated